runtime/vm/VMAccess.cpp

```diff
@@ -598,9 +598,17 @@ void releaseExclusiveVMAccess(J9VMThread * vmThread)
 			PORT_ACCESS_FROM_JAVAVM(vm);
 			j9mem_free_memory(currentThread->lastDecompilation);
 			currentThread->lastDecompilation = NULL;
+			if (NULL != currentThread->utfCache) {
```

Comment:
Would it be better to dump the cache on exclusive acquire?  We'd potentially GC the cache items earlier then or are we just letting the cache hold stale values between exclusive acquire and release?  That may be confusing to Service

Reply:
It could be, but the other freeing code was already here. If we change to use the GC hook, I would dump it on GC start. The cache is never read or written by the GC.